aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -46,6 +52,309 @@ bool should_use_th_sort(const Tensor &self, int64_t dim) {
 
 std::vector<int64_t> infer_dense_strides_dim_last(const Tensor & self, int64_t dim);
 
+void fillSliceWithIndex(Tensor& t,
+                        int dim) {
+  int64_t dims = t.dim();
```

Comment:
this whole fillSliceWithIndex can be replaced with the following:
```
auto sizes = DimVector(t.ndim(), 1);
sizes[dim]=t.sizes()[dim];
auto range = torch::arange(t.sizes()[dim], t.options());
auto rangeview = range.view(sizes).expans_as(t);
t.copy_(rangeview);
```
It is marginally slower for sorting along last dimension (because there's additional arange kernel), but it is faster for other cases, and doesn't rely on legacy kernels.